scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/ServerHandshaker.java

```diff
@@ -717,16 +739,17 @@ private void negotiateCipherSuite(final ClientHello clientHello, final HelloExte
 
 		CertificateType supportedServerCertType = getSupportedServerCertificateType(clientHello);
 		CertificateType supportedClientCertType = getSupportedClientCertificateType(clientHello);
-		SupportedGroup group = negotiateNamedCurve(clientHello);
+		List<SupportedGroup> groups = negotiateNamedCurves(clientHello);
 		List<SignatureAndHashAlgorithm> signatures = negotiateSignatureAndHashAlgorithms(clientHello);
+		ECPointFormat format = negotiateECPointFormat(clientHello);
 
 		for (CipherSuite cipherSuite : clientHello.getCipherSuites()) {
 			// NEVER negotiate NULL cipher suite
 			if (cipherSuite != CipherSuite.TLS_NULL_WITH_NULL_NULL && supportedCipherSuites.contains(cipherSuite)) {
-				if (isEligible(cipherSuite, supportedServerCertType, supportedClientCertType, group, signatures)) {
+				if (isEligible(cipherSuite, supportedServerCertType, supportedClientCertType, groups, format, signatures)) {
 					negotiatedServerCertificateType = supportedServerCertType;
 					negotiatedClientCertificateType = supportedClientCertType;
-					negotiatedSupportedGroup = group;
+					negotiatedSupportedGroups = groups;
```

Comment:
why negociateGroups is a list ?

Reply:
If list or single value depends also on the extensions.
For some the client sends a list, and the server the selected item.
For others the client sends a list, and the server sends the selected sublist.

Reply:
currently the code only use "the selected item" (1 group)
So this list is just for the future ? (or I missed something?)

Reply:
Exactly, currently **one** is selected, because the server hello extensions are limited to **one** and this implementation is mainly from the time before adding the signing stuff and evaluating the curves!
With #1259 I would change that to:
common parameter: lists
negotiated parameter: lists and selected items (depending on the specs and usage).